salt/_modules/metalk8s_network.py

```diff
@@ -229,7 +229,7 @@ def routes():
 
 
 def get_control_plane_ingress_ip():
-    if "ingress" in __pillar__["networks"]["control_plane"]:
+    if __pillar__["networks"]["control_plane"].get("ingress", {}).get("ip"):
         return __pillar__["networks"]["control_plane"]["ingress"]["ip"]
```

Comment:
Could be simplified with a try/except:

```suggestion
    try:
        return __pillar__["networks"]["control_plane"]["ingress"]["ip"]
    except KeyError:
        pass
```

Reply:
Hmm not exactly the same for example if you set the IP to `None` but right we may not want to support this :smiley: